crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1337,8 +1337,7 @@ impl<'db> TypeInferenceBuilder<'db> {
         &mut self,
         _literal: &ast::ExprEllipsisLiteral,
     ) -> Type<'db> {
-        // TODO Ellipsis
-        Type::Unknown
+        Type::EllipsisType
```

Comment:
I think I would probably just revert the changes you made to `types.rs` and `types/display.rs` and simply do this here:

```suggestion
        builtins_symbol_ty_by_name(self.db, "Ellipsis")
```

It doesn't seem to get us quite the right answer currently (the typeshed stubs there are using some features we don't support right now) so you may need to adjust the assertion in your test and add a TODO. But it'll set us up better in the long run.

Reply:
That seems reasonable. The resulting display/result is `Unknown | Literal[EllipsisType]`. As I am leaving a TODO: what *should* it be?

Reply:
It should just be `types.EllipsisType` (if on py310+) or `ellipsis` (if on Python <=3.9)